pinot-common/src/main/java/org/apache/pinot/common/metrics/ServerGauge.java

```diff
@@ -91,8 +91,9 @@ public enum ServerGauge implements AbstractMetrics.Gauge {
   LUCENE_INDEXING_DELAY_MS("milliseconds", false),
   LUCENE_INDEXING_DELAY_DOCS("documents", false),
   // Needed to track if valid doc id snapshots are present for faster restarts
-  UPSERT_VALID_DOC_ID_SNAPSHOT_COUNT("upsertValidDocIdSnapshotCount", false),
+  UPSERT_SNAPSHOT_COUNT("upsertSnapshotCount", false),
```

Comment:
can we avoid changing the metric here and in ServerMeter ? it becomes backward incompatible for users using the current metric